packages/xo-web/src/xo-app/new/network/index.js

```diff
@@ -0,0 +1,276 @@
+  })),
+  injectIntl,
+  provideState({
+    initialState: getInitialState,
+    effects: {
+      linkState,
+      onChangeBondedNetwork: () => ({ bondedNetwork }) => ({
+        ...getInitialState(),
+        bondedNetwork: !bondedNetwork,
+      }),
+      onChangePif: (_, value) => ({ bondedNetwork }) =>
+        bondedNetwork ? { pifs: value } : { pif: value },
+      reset: getInitialState,
+    },
+    computed: {
+      pifPredicate: (_, { pool }) => pif =>
+        pif.vlan === -1 && pif.$host === (pool && pool.master),
+    },
+  }),
+  injectState,
+  class extends Component {
+    static contextTypes = {
+      router: PropTypes.object,
+    }
+
+    componentWillMount() {
+      getBondModes().then(bondModes => this.setState({ bondModes }))
+    }
+
+    componentWillReceiveProps(props) {
+      this.forceUpdate()
+    }
+
+    _create = () => {
+      const { bondModes } = this.state
+      const { pool, state } = this.props
+      const {
+        bondedNetwork,
+        bondMode,
+        description,
+        mtu,
+        name,
+        pif,
+        pifs,
+        vlan,
+      } = state
+      return bondedNetwork
+        ? createBondedNetwork({
+            bondMode: bondMode || bondModes[0],
+            description,
+            mtu,
+            name,
+            pifs: map(pifs, pif => pif.id),
+            pool: pool.id,
+            vlan,
+          })
+        : createNetwork({
+            description,
+            mtu,
+            name,
+            pif: pif.id,
+            pool: pool.id,
+            vlan,
+          })
+    }
+
+    _selectPool = pool => {
+      const { pathname } = this.props.location
+      this.context.router.push({
+        pathname,
+        query: pool && { pool: pool.id },
+      })
+    }
+
+    _redirectOnSuccess = network =>
+      network !== undefined ? `pools/${this.props.pool.id}/network` : undefined
+
+    _renderHeader = () => {
+      const { isAdmin, isPoolAdmin, pool } = this.props
+      return (
+        <Container>
+          <Row>
+            <Col mediumSize={12}>
+              <h2>
+                {isAdmin || (isPoolAdmin && process.env.XOA_PLAN > 3)
+                  ? _('createNewNetworkOn', {
+                      select: (
+                        <span className={styles.inlineSelect}>
+                          <SelectPool
+                            onChange={this._selectPool}
+                            value={pool}
+                          />
+                        </span>
+                      ),
+                    })
+                  : _('createNewNetworkNoPermission')}
+              </h2>
+            </Col>
+          </Row>
+        </Container>
+      )
+    }
+
+    render() {
+      const { bondModes } = this.state
+      const { state, effects, intl, pool } = this.props
+      const {
+        bondedNetwork,
+        bondMode = bondModes !== undefined ? bondModes[0] : undefined,
```

Comment:
This is weird. You can't show something in the UI if it doesn't match the `state` value.